### ISSUE TITLE
Large File Upload/Download

### ISSUE THREAD

**Issue by NickEckert** [restfb/restfb#319]
I'm trying to deploy and application that uses restFB on a server where I accept user requests to upload/download files. As it stands I have to store the entire file, no matter the size, in memory before It's uploaded/downloaded. Is there a better way of approaching this using restFB without having to write my own http calls, or maybe I'm missing something?

**Comment by nbartels** [comment]
You can provide an InputStream as input for the BinaryAttachment and the data is fetched on upload. Even though this method is deprecated you may use it. 

Because of the increasing filesize we have to find a good way to support this and have a small memory footprint.

Can you please tell me what file size we are talking about, because I'd like to write some integration test to solve this issue :smile:

**Comment by NickEckert** [comment]
Well it would be nice to support up to the Facebook file size limit which is 1G for videos unless resumable upload was supported in which case it would be 1.75G or 45 minutes in length. I was trying to upload a 25MB video file the other day using a byte array as the binary attachment and was receiving timeouts from Facebook sporadically.

**Comment by nbartels** [comment]
Okay I see. I don't see a direct support for downloads. You get a URL from Facebook and have to fetch the file on your own. I don't think it makes any sense to encapsulate a FileDownloader in RestFB. 

Upload is an other topic. Did you try the InputStream with the BinaryAttachment. 

The resumable Uploads are still a open task and we have to check how we can integrate it in a way it's easy usable but still flexible.

**Comment by nbartels** [comment]
The resumable upload topic is more a documentation issue. I wrote a little tool to upload big files to Facebook using the resumable upload and RestFB. I added some response type to make the process a bit more easy, but in general it's without problems possible to use the new kind of upload.

I let this issue open until the resumable upload documentation is online, so it is trackable that there's still  something to do.
